gapis/api/vulkan/transform_command_disabler.go

```diff
@@ -280,7 +302,41 @@ func (disablerTransform *commandDisabler) rewriteCommandBuffer(ctx context.Conte
 	return newCmdBuffer, nil
 }
 
-func (disablerTransform *commandDisabler) getNewCommandBufferAndBegin(ctx context.Context, cmd *VkQueueSubmit, inputState *api.GlobalState) (VkCommandBuffer, error) {
+func (disablerTransform *commandDisabler) rewriteExecuteSecondaryCommandBuffer(ctx context.Context,
+	idx api.SubCmdIdx, primaryCommandBuffer VkCommandBuffer, args VkCmdExecuteCommandsArgsʳ,
+	cmd *VkQueueSubmit, inputState *api.GlobalState) (api.Cmd, error) {
+
+	existingCmdBufferCount := uint32(args.CommandBuffers().Len())
+	newCommandBuffers := make([]VkCommandBuffer, 0, existingCmdBufferCount)
+
+	for i := uint32(0); i < existingCmdBufferCount; i++ {
+		currentSubCmdID := append(idx, uint64(i))
+		existingCommandBuffer := args.CommandBuffers().Get(uint32(i))
+		if !disablerTransform.doesContainDisabledCmd(currentSubCmdID) {
+			newCommandBuffers = append(newCommandBuffers, existingCommandBuffer)
+			continue
+		}
+
+		existingCommandBufferObject := GetState(inputState).CommandBuffers().Get(existingCommandBuffer)
+		newCommandBuffer, err := disablerTransform.rewriteCommandBuffer(ctx, currentSubCmdID, existingCommandBufferObject, cmd, inputState)
+		if err != nil {
+			return nil, fmt.Errorf("Error during writing secondary command buffer : %v", err)
```

Comment:
When there is a context and you want to propagate an error up with an additional message, prefer using log.Err when possible, e.g.: `return nil, log.Err(ctx, err, "Failed to rewrite secondary command buffer")`